python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -275,17 +275,25 @@ def get_asset_materializations(
     check.opt_mapping_param(tags, "tags", key_type=str, value_type=str)
 
     instance = graphene_info.context.instance
-    event_records = instance.get_event_records(
-        EventRecordsFilter(
-            event_type=DagsterEventType.ASSET_MATERIALIZATION,
-            asset_key=asset_key,
-            asset_partitions=partitions,
-            before_timestamp=before_timestamp,
-            after_timestamp=after_timestamp,
-            tags=tags,
-        ),
-        limit=limit,
-    )
+    cursor = None
+    event_records = []
+    while True:
```

Comment:
is there a way to enforce that all the callers have a non-null `limit` set?

it looks like the main call site without a limit is this `resolve_latestMaterializationByPartition` field.  I feel like we could probably switch this to use `get_latest_storage_id_by_partition` to fetch the ids and then another call to `fetch_materializations` using a storage_id filter...

Reply:
@prha latestMaterializationByPartition is the specific field we're trying to optimize actually - if that callsite passes in, say, 27k storage IDs, would you expect us to be able to query them 10k a time without timing out? I'm seeing timeouts when I try but i'm not sure if that's related to my test dagit setup

Reply:
yeah, i'd recommend using `get_latest_storage_id_by_partition` route there, which has an optimized implementation in cloud.

Reply:
right - i was referring to after get_latest_storage_id_by_partition returns the 27k storage ids and we are trying to fetch them by ID - i was finding that fetching those 10k events by storage ID was timing out

Reply:
oh, I see.  yeah, maybe we need to refine the signatures to handle that level of pagination?